### PR TITLE
adding a mini benchmark for cross entropy loss

### PR DESCRIPTION
**Pull request by protonu**

### CODE REVIEW

File: benchmarks/python/test_cross_entropy_loss.py [hunk 1]
```diff
@@ -80,3 +83,49 @@ def fwd_call(inp):
...
+    batch_size = 4096
+
+    def fwd_call(inp):
+        return SyntheticMiniModel.mini_model(*inp)
+
+    input, target = SyntheticMiniModel.inputs(batch_size, vocab_size)
+
+    kwargs = {}
+    inputs = [input, target]
+    fwd_fn = with_executor(executor, fwd_call, **kwargs)
```
Comment (jjsjann123) [inline benchmarks/python/test_cross_entropy_loss.py:106]: nitpick, you don't need to pass an empty kwargs.

File: benchmarks/python/test_cross_entropy_loss.py [hunk 2]
```diff
@@ -80,3 +83,49 @@ def fwd_call(inp):
...
+    outputs = fwd_fn(inputs)
+    grads = SyntheticMiniModel.grads()
+    run_benchmark(benchmark, unary_bwd_torch, [outputs, grads, inputs])
```
Comment (jjsjann123) [inline benchmarks/python/test_cross_entropy_loss.py:131]: I think you need to flatten `inputs` here.
Since our io computation doesn't do tree flatten on inputs.

File: benchmarks/python/core.py [hunk 3]
```diff
@@ -164,9 +164,13 @@ def set_metrics(
             for inp in inputs:
                 if isinstance(inp, torch.Tensor):
                     iobytes += inp.element_size() * inp.numel()
-            for out in outputs:
-                if isinstance(out, torch.Tensor):
-                    iobytes += out.element_size() * out.numel()
+
+            if isinstance(outputs, torch.Tensor) and outputs.dim() == 0:
```
[the remark this answers is not in the document]
Comment (protonu) [reply benchmarks/python/core.py:168]: I am not sure if that's quite right.
'''
a = torch.tensor(5)
print(isinstance(a, Iterable))
'''
This should be true.
So outputs could be a single torch tensor and not get wrapped. Please let me know if I am mistaken.

File: benchmarks/python/core.py [hunk 4]
```diff
@@ -164,9 +164,13 @@ def set_metrics(
...
+                iobytes += outputs.element_size() 
+            else:    
+                for out in outputs:
+                    if isinstance(out, torch.Tensor):
+                        iobytes += out.element_size() * out.numel()
```
[the remark this answers is not in the document]
Comment (protonu) [reply benchmarks/python/core.py]: Please let me know if we still need this.

File: benchmarks/python/test_cross_entropy_loss.py (hunk 2 above)
Comment (protonu) [reply benchmarks/python/test_cross_entropy_loss.py:127]: Can you explain this a bit - I wasn't quite sure what it meant.

File: benchmarks/python/core.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (jjsjann123) [reply benchmarks/python/core.py]: Oh you are right and I didn't know that!

@Priya2698 so it looks like line 158 - 161 isn't doing the right thing?

File: benchmarks/python/core.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (jjsjann123) [reply benchmarks/python/core.py]: The initial comment still holds I think
i.e. if `out` is a scalar tensor (dim() == 0), we shouldn't multiple its numel

[the remark this answers is not in the document]
Comment (protonu) [reply benchmarks/python/core.py]: In this case isn't `numel()` just 1? Maybe I have this wrong - I thought even in this case we will get 1 * `element_size()`?

**Review by jjsjann123 — APPROVED** [review]
LGTM

**Comment by protonu** [comment]
!test

File: python/nvfuser_direct/_C_DIRECT.cpython-312-x86_64-linux-gnu.so
Comment (crcrpar) [inline python/nvfuser_direct/_C_DIRECT.cpython-312-x86_64-linux-gnu.so:1]: Would this dynamic library seriously need to be in this directory?

Comment (protonu) [reply python/nvfuser_direct/_C_DIRECT.cpython-312-x86_64-linux-gnu.so:1]: Thanks for catching that! That was a mistake
